lisa/tools/make.py

```diff
@@ -24,14 +30,32 @@ def _install(self) -> bool:
         posix_os.install_packages([self, Gcc])
         return self._check_exists()
 
-    def make_and_install(self, cwd: PurePath) -> None:
-        # make/install can happen on different folder with same parameter,
-        # so force rerun it.
-        make_result = self.run(force_run=True, shell=True, cwd=cwd)
-        if make_result.exit_code == 0:
-            # install with sudo
-            self.node.execute("make install", shell=True, sudo=True, cwd=cwd)
-        else:
-            raise LisaException(
-                f"'make' command got non-zero exit code: {make_result.exit_code}"
-            )
+    def make_install(self, cwd: PurePath, timeout: int = 600) -> None:
+        self.make(arguments="", cwd=cwd, timeout=timeout)
+
+        # install with sudo
+        self.make(arguments="install", cwd=cwd, sudo=True, timeout=timeout)
+
+    def make(  # type: ignore
```

Comment:
Why ignore? And why that piping of yes to it?

Reply:
Good catch, not sure how it happens. Removed